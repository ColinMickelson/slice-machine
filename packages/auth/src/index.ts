export { createPrismicAuth } from "./createPrismicAuth";

export { createPrismicAuthMiddleware } from "./createPrismicAuthMiddleware";

export type { PrismicAuth } from "./PrismicAuth";

export type { PrismicUserProfile } from "./types";

export {
	UnauthenticatedError,
	isUnauthenticatedError,
	UnexpectedDataError,
	isUnexpectedDataError,
} from "./errors";
