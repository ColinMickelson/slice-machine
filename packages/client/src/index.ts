import axios, { AxiosError, AxiosPromise, AxiosRequestHeaders } from "axios";
import * as t from "io-ts";

import { UserProfile } from "@slicemachine/core/build/models";
import { SharedSlice } from "@prismicio/types-internal/lib/customtypes/widgets/slices";
import { CustomType } from "@prismicio/types-internal/lib/customtypes/CustomType";

import type { ClientError, UploadParameters } from "./models";
import {
  ApplicationMode,
  getStatus,
  getMessage,
  ApisEndpoints,
  ProductionApisEndpoints,
  StageApisEndpoints,
  AclCreateResult,
} from "./models";

import { getAndValidateResponse } from "./utils";
import { upload } from "./utils/upload";

// exporting models to be used with the Client.
export * from "./models";

// exporting utils so the client calls can be replicated for speficic calls to a consumer.
export * from "./utils";

// exporting the client itself.
export class Client {
  apisEndpoints: ApisEndpoints;
  repository: string | null;
  authenticationToken: string;

  constructor(
    applicationMode: ApplicationMode,
    repository: string | null,
    authenticationToken: string
  ) {
    this.repository = repository;
    this.authenticationToken = authenticationToken;

    if (applicationMode === ApplicationMode.PROD)
      this.apisEndpoints = ProductionApisEndpoints;
    else if (applicationMode === ApplicationMode.STAGE)
      this.apisEndpoints = StageApisEndpoints;
    else {
      // Dev
      this.apisEndpoints = ProductionApisEndpoints;
    }
  }

  // setters to provide flexibility
  updateAuthenticationToken(newToken: string): Client {
    this.authenticationToken = newToken;
    return this
  }

  updateRepository(repository: string | null): Client {
    this.repository = repository;
    return this
  }

  // private methods
  _fetch(
    method: "get" | "post",
    url: string,
    data?: Record<string, unknown>,
    headers?: AxiosRequestHeaders
  ): AxiosPromise {
    return axios({
      method,
      url,
      data,
      headers: {
        Authorization: `Bearer ${this.authenticationToken}`,
        "User-Agent": "slice-machine",
        ...(this.repository ? { repository: this.repository } : {}),
        ...headers,
      },
    });
  }

  _get(url: string): AxiosPromise {
    return this._fetch("get", url);
  }

  _post(url: string, data: Record<string, unknown>): AxiosPromise {
    return this._fetch("post", url, data).catch((error: Error | AxiosError) => {
      const status: number = getStatus(error);
      const message: string = getMessage(error);

      // Making sure the error is typed
      const clientError: ClientError = { status, message };
      return Promise.reject(clientError);
    });
  }

  async validateAuthenticationToken(): Promise<boolean> {
    return this._get(
      `${this.apisEndpoints.Authentication}validate?token=${this.authenticationToken}`
    )
      .then(() => true)
      .catch((error: Error | AxiosError) => {
        const status: number = getStatus(error);
        if (status == 401) return false;

        const message: string = getMessage(error);

        // Making sure the error is typed
        const clientError: ClientError = { status, message };
        return Promise.reject(clientError);
      });
  }

  async refreshAuthenticationToken(): Promise<string> {
    return getAndValidateResponse<string>(
      this._get(
        `${this.apisEndpoints.Authentication}refreshtoken?token=${this.authenticationToken}`
      ),
      "refreshed authentication token",
      t.string
    ).then((newToken) => {
      this.updateAuthenticationToken(newToken);
      return newToken;
    });
  }

  async profile(): Promise<UserProfile> {
    return getAndValidateResponse<UserProfile>(
      this._get(`${this.apisEndpoints.Users}profile`),
      "user profile",
      UserProfile
    );
  }

  async getSlices(): Promise<SharedSlice[]> {
    return getAndValidateResponse<SharedSlice[]>(
      this._get(`${this.apisEndpoints.Models}slices`),
      "shared slices",
      t.array(SharedSlice)
    );
  }

  async getCustomTypes(): Promise<CustomType[]> {
    return getAndValidateResponse<CustomType[]>(
      this._get(`${this.apisEndpoints.Models}customtypes`),
      "custom types",
      t.array(CustomType)
    );
  }

  async insertCustomType(customType: CustomType): Promise<AxiosPromise> {
    return this._post(
      `${this.apisEndpoints.Models}customtypes/insert`,
      customType
    );
  }

  async updateCustomType(customType: CustomType): Promise<AxiosPromise> {
    return this._post(
      `${this.apisEndpoints.Models}customtypes/update`,
      customType
    );
  }

  async insertSlice(sharedSlice: SharedSlice): Promise<AxiosPromise> {
    return this._post(`${this.apisEndpoints.Models}slices/insert`, sharedSlice);
  }

  async updateSlice(sharedSlice: SharedSlice): Promise<AxiosPromise> {
    return this._post(`${this.apisEndpoints.Models}slices/update`, sharedSlice);
  }

  async createImagesAcl(): Promise<AclCreateResult> {
    return getAndValidateResponse<AclCreateResult>(
      this._get(`${this.apisEndpoints.AclProvider}create`),
      "acl",
      AclCreateResult
    );
  }

  async deleteImagesFolderAcl(sliceName: string): Promise<AxiosPromise> {
    return this._post(`${this.apisEndpoints.AclProvider}delete-folder`, {
      sliceName,
    });
  }

  async uploadImageAcl(params: UploadParameters): Promise<number | undefined> {
    return upload(params);
  }
}
