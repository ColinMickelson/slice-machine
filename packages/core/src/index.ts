export * as FileSystem from "./filesystem";
export * as Utils from "./utils";
export * from "./core";
export * as Models from "./models";
export * as Libraries from "./libraries";
export * as Mocks from "./libraries/mocks";
export * from "./auth";
