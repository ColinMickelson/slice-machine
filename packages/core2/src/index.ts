export {
	SliceMachineManager,
	createSliceMachineManager,
} from "./createSliceMachineManager";

export {
	CreateSliceMachineManagerMiddlewareArgs,
	createSliceMachineManagerMiddleware,
} from "./createSliceMachineManagerMiddleware";

export {
	PrismicAuthManager,
	PrismicUserProfile,
	createPrismicAuthManager,
} from "./createPrismicAuthManager";

export {
	CreatePrismicAuthManagerMiddlewareArgs,
	PrismicAuthCheckStatusResponse,
	createPrismicAuthManagerMiddleware,
} from "./createPrismicAuthManagerMiddleware";
