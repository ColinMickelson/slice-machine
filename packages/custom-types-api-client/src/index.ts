export { createCustomTypesAPIClient } from "./createCustomTypesAPIClient";

export type { CustomTypesAPIClient } from "./CustomTypesAPIClient";

export {
	InvalidPayloadError,
	ForbiddenError,
	ConflictError,
	NotFoundError,
	PrismicError,
} from "@prismicio/custom-types-client";
