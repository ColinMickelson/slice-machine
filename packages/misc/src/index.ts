// TODO: Figure out where libs should be
export { decode } from "./decode";
export type { DecodeReturnType } from "./decode";
export { DecodeError } from "./DecodeError";
export {
	readPrismicrc,
	writePrismicrc,
	updatePrismicrc,
	Prismicrc,
} from "./prismicrc";
export {
	getAllStableSliceMachineVersions,
	getReleaseNotesForVersion,
} from "./versions";

// Errors
export { SliceMachineError, isSliceMachineError } from "./SliceMachineError";
export { InternalError, isInternalError } from "./InternalError";

// Constants
export { ApplicationMode } from "./ApplicationMode";
export { APIEndpoints } from "./APIEndpoints";
export { APITokens } from "./APITokens";
