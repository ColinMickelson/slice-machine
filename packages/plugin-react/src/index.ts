import type {
  Plugin,
  FilenameAndData,
  Variations,
} from "@slicemachine/plugin-middleware";
import { FieldType } from "@slicemachine/plugin-middleware";

export const framework: Plugin["framework"] = "react";

export const slice = (name: string): FilenameAndData<string> => ({
  filename: "index.js",
  data: `import React from 'react'
import { RichText } from 'prismic-reactjs'
  
const ${name} = ({ slice }) => (
  <section>
    <span className="title">
      { slice.primary.title ? (
        <RichText render={slice.primary.title} />
      ) : (
        <h2>Template slice, update me!</h2>
      )}
    </span>
      { slice.primary.description ? (
        <RichText render={slice.primary.description}/>
      ) : (
         <p>start by editing this slice from inside Slice Machine!</p>
      )}
      <style jsx>{\`
          section {
            max-width: 600px;
            margin: 4em auto;
            text-align: center;
          }
          .title {
            color: #8592e0;
          }
      \`}</style>
    </section>
  )
  
  export default ${name}
`,
});

function safeComponentName(name: string): string {
  return `_${name.replace(/(^\w|-+\w)/g, (text) =>
    text.replace(/-/g, "").toUpperCase()
  )}`;
}

export const story = (
  path: string,
  sliceName: string,
  variations: Variations
): FilenameAndData<string> => {
  const exportStatements = variations
    .map((variation) => {
      const name = safeComponentName(variation.id);
      return `
export const ${name} = () => <MyComponent slice={${JSON.stringify(
        variation
      )}} />
${name}.storyName = '${variation.name}'
`;
    })
    .join("/n");

  return {
    filename: "index.stories.js",
    data: `import MyComponent from '${path}';
export default { title: '${sliceName}' }
${exportStatements}
`,
  };
};

export const index = (slices: Array<string>): FilenameAndData<string> => ({
  filename: "index.js",
  data: `// This file is generated by Slice Machine. Update with care!

${slices
  .map((slice) => `export { default as ${slice} } from './${slice}'`)
  .join("\n")}
`,
});

export const snippets = ({
  type,
  fieldText,
  useKey = false,
  isRepeatable = false,
}: {
  type: FieldType;
  fieldText: string;
  useKey?: boolean;
  isRepeatable?: boolean;
}): string => {
  if (isRepeatable) {
    const code = snippets({ type, fieldText, useKey: true });
    return `{ slice?.items?.map((item, i) => ${code}) }`;
  }
  switch (type) {
    case FieldType.Boolean:
      return `<span ${
        useKey ? `key="bool-\${i}"` : ""
      }> { ${fieldText} ? 'true' : 'false' }</span>`;

    case FieldType.Color:
      return `<span  ${
        useKey ? `key="color-\${i}"` : ""
      } style={{ color: ${fieldText} }}>Some Text</span>`;

    case FieldType.ContentRelationship:
      return `/* import { Link } from 'prismic-reactjs' */
<a ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      } href={Link.url(${fieldText})}>My Link</a>`;

    case FieldType.Date:
      return `/* import { Date as ParseDate } from 'prismic-reactjs' */
<span ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      }>{ ParseDate(${fieldText}) }</span>`;

    case FieldType.Embed:
      return `// you might want to use a lib here (eg. react-oembed-container)
<div ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      }} dangerouslySetInnerHTML={{ __html: ${fieldText} }} />`;

    case FieldType.GeoPoint:
      return `<span ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      }>{ ${fieldText} }</span>`;

    case FieldType.Group:
      return "";

    case FieldType.Image:
      return `<img src={${fieldText}.url} alt={${fieldText}.alt} ${
        useKey ? `key="img-\${i}"` : ""
      }/>`;

    case FieldType.IntegrationFields:
      return "";

    case FieldType.Link:
      return `/* import { Link } from 'prismic-reactjs' */
<a ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      } href={Link.url(${fieldText})}>My Link</a>`;

    case FieldType.LinkToMedia:
      return `/* import { Link } from 'prismic-reactjs' */
<a ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      } href={Link.url(${fieldText})}>My Link</a>`;

    case FieldType.Number:
      return `<span ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      }>{ ${fieldText} }</span>`;

    case FieldType.Select:
      return `<span ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      }>{ ${fieldText} }</span>`;

    case FieldType.StructuredText:
      return `<RichText render={${fieldText}} ${
        useKey ? `key="rich-text-\${i}"` : ""
      } />`;

    case FieldType.Text:
      return `<span ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      }>{ ${fieldText} }</span>`;

    case FieldType.Timestamp:
      return `<span ${
        useKey ? `key="${fieldText}-\${i}"` : ""
      }>{ ${fieldText} }>/span>`;

    case FieldType.UID:
      return `<span>{ ${fieldText} }</span>`;
    default:
      return "";
  }
};
