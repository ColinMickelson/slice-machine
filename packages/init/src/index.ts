#!/usr/bin/env node

import { Utils, FileSystem } from "@slicemachine/core";
import { Tracker } from "./utils/tracker";
import {
  installRequiredDependencies,
  validatePkg,
  maybeExistingRepo,
  createRepository,
  loginOrBypass,
  configureProject,
  displayFinalMessage,
  detectFramework,
  installLib,
} from "./steps";
import { findArgument } from "./utils";

async function init() {
  const cwd = findArgument(process.argv, "cwd") || process.cwd();
  const base = findArgument(process.argv, "base") || Utils.CONSTS.DEFAULT_BASE;
  const lib: string | undefined = findArgument(process.argv, "library");
  const branch: string | undefined = findArgument(process.argv, "branch");
  const tracking =
    findArgument(process.argv, "tracking") === "false" ? false : true;

  console.log(
    Utils.purple(
      "You're about to configure Slicemachine... Press ctrl + C to cancel"
    )
  );

  // verify package.json file exist
  validatePkg(cwd);

  // login
  const user = await loginOrBypass(base);
  if (!user) throw new Error("The user should be logged in!");

  // retrieve tokens for api calls
  const config = FileSystem.PrismicSharedConfigManager.get();

  // detect the framework used by the project
  const frameworkResult = await detectFramework(cwd);

  // select the repository used with the project.
  const { existing, name } = await maybeExistingRepo(
    config.cookies,
    cwd,
    config.base
  );

  if (!existing) {
    await createRepository(name, frameworkResult.value, config);
  }

  const tracker = Tracker.build(
    "JfTfmHaATChc4xueS7RcCBsixI71dJIJ",
    name,
    {
      userId: user.userId,
    },
    tracking
  );

  // install the required dependencies in the project.
  await installRequiredDependencies(cwd, frameworkResult.value);

  const sliceLibPath = lib
    ? await installLib(tracker, cwd, lib, branch)
    : undefined;

  // configure the SM.json file and the json package file of the project..
  configureProject(cwd, base, name, frameworkResult, sliceLibPath, tracking);

  // ask the user to run slice-machine.
  displayFinalMessage(cwd);
}

try {
  void init();
} catch (error) {
  if (error instanceof Error) Utils.writeError(error.message);
  else console.error(error);
}
