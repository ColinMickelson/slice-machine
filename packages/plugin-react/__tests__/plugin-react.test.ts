import { expect, test } from "@jest/globals";
import * as plugin from "../src";
import type { Variations } from "@slicemachine/plugin-middleware";
import { FieldType } from "@slicemachine/plugin-middleware";

describe("@slicemachine/plugin-react", () => {
  test("#framework", () => {
    expect(plugin.framework).toEqual("react");
  });
  test("#slice", () => {
    const result = plugin.slice("Foo");
    expect(result.data).toContain("const Foo = ({ slice }) => (");
    expect(result.data).toContain("export default Foo");
    expect(result).toMatchSnapshot();
  });

  test("#story", () => {
    const variations = [
      {
        id: "default-slice",
        name: "Default slice",
        docURL: "...",
        version: "sktwi1xtmkfgx8626",
        description: "FooSlice",
        primary: {
          title: {
            type: "StructuredText" as const,
            config: {
              single: "heading1",
              label: "Title",
              placeholder: "This is where it all begins...",
            },
          },
          description: {
            type: "StructuredText",
            config: {
              single: "paragraph",
              label: "Description",
              placeholder: "A nice description of your feature",
            },
          },
          "": {
            config: {
              label: "",
              constraint: {},
              thumbnails: [],
            },
            type: "Image",
          },
        },
        items: {},
      },
    ] as Variations;
    const result = plugin.story("../slices/foo", "FooBar", variations);
    expect(result.filename).toEqual("index.stories.js");
    expect(result.data).toContain("import MyComponent from '../slices/foo'");

    expect(result.data).toContain("export default { title: 'FooBar' }");

    expect(result.data).toContain(
      `export const _DefaultSlice = () => <MyComponent slice={`
    );

    expect(result.data).toContain(
      `_DefaultSlice.storyName = '${variations[0].name}'`
    );

    expect(result).toMatchSnapshot();
  });

  describe("#index", () => {
    test("it creates an index.js file", () => {
      const result = plugin.index(["Foo", "Bar", "Baz"]);
      expect(result.filename).toEqual("index.js");
      expect(result.data).toContain("export { default as Foo } from './Foo'");
      expect(result.data).toMatchSnapshot();
    });
  });

  describe("#snippets", () => {
    const SupportedFields = Object.values(FieldType).filter(
      (type) => type !== FieldType.Group && type !== FieldType.IntegrationFields
    );

    const suported = SupportedFields.map((field, i) => [
      field,
      `slices[${i}].data`,
    ]);

    test.each(suported)(
      "supported type %s should be truthy",
      (type, fieldText) => {
        const result = plugin.snippets({ type, fieldText });
        expect(result).toBeTruthy();
        expect(result).toMatchSnapshot();
      }
    );

    test.each(suported.map((d) => [...d, true, false]))(
      "snippet %s with useKey",
      (type, fieldText, useKey, isRepeatable) => {
        const result = plugin.snippets({
          type,
          fieldText,
          useKey,
          isRepeatable,
        });
        expect(result).toBeTruthy();
        expect(result).toMatchSnapshot();
      }
    );

    const UnsupprtedFields = Object.values(FieldType).filter(
      (type) => SupportedFields.includes(type) === false
    );

    const unsupported = UnsupprtedFields.map((field, i) => [
      field,
      `slice[${i}].data`,
    ]);

    test.each(unsupported)(
      "unsuported types %s should be falsy",
      (type, fieldText) => {
        expect(plugin.snippets({ type, fieldText })).toBeFalsy();
      }
    );

    test("repeatable slices", () => {
      const result = plugin.snippets({
        type: FieldType.Text,
        fieldText: "slices",
        useKey: false,
        isRepeatable: true,
      });

      expect(result).toEqual(
        '{ slice?.items?.map((item, i) => <span key="slices-${i}">{ slices }</span>) }'
      );
    });
  });
});
