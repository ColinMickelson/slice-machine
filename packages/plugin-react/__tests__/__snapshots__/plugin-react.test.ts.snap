// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@slicemachine/plugin-react #index it creates an index.js file 1`] = `
"// This file is generated by Slice Machine. Update with care!

export { default as Foo } from './Foo'
export { default as Bar } from './Bar'
export { default as Baz } from './Baz'
"
`;

exports[`@slicemachine/plugin-react #slice 1`] = `
Object {
  "data": "import React from 'react'
import { RichText } from 'prismic-reactjs'
  
const Foo = ({ slice }) => (
  <section>
    <span className=\\"title\\">
      { slice.primary.title ? (
        <RichText render={slice.primary.title} />
      ) : (
        <h2>Template slice, update me!</h2>
      )}
    </span>
      { slice.primary.description ? (
        <RichText render={slice.primary.description}/>
      ) : (
         <p>start by editing this slice from inside Slice Machine!</p>
      )}
      <style jsx>{\`
          section {
            max-width: 600px;
            margin: 4em auto;
            text-align: center;
          }
          .title {
            color: #8592e0;
          }
      \`}</style>
    </section>
  )
  
  export default Foo
",
  "filename": "index.js",
}
`;

exports[`@slicemachine/plugin-react #snippets snippet Boolean with useKey 1`] = `"<span key=\\"bool-\${i}\\" >{ slices[0].data ? 'true' : 'false' }</span>"`;

exports[`@slicemachine/plugin-react #snippets snippet Color with useKey 1`] = `"<span key=\\"color-\${i}\\" style={{ color: slices[1].data }}>Some Text</span>"`;

exports[`@slicemachine/plugin-react #snippets snippet ContentRelationship with useKey 1`] = `
"/* import { Link } from 'prismic-reactjs' */
<a key=\\"slices[2].data-\${i}\\" href={Link.url(slices[2].data)}>My Link</a>"
`;

exports[`@slicemachine/plugin-react #snippets snippet Date with useKey 1`] = `
"/* import { Date as ParseDate } from 'prismic-reactjs' */
<span key=\\"slices[3].data-\${i}\\">{ ParseDate(slices[3].data) }</span>"
`;

exports[`@slicemachine/plugin-react #snippets snippet Embed with useKey 1`] = `
"// you might want to use a lib here (eg. react-oembed-container)
<div key=\\"slices[4].data-\${i}\\" dangerouslySetInnerHTML={{ __html: slices[4].data }} />"
`;

exports[`@slicemachine/plugin-react #snippets snippet GeoPoint with useKey 1`] = `"<span key=\\"slices[5].data-\${i}\\" >{ slices[5].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets snippet Image with useKey 1`] = `"<img src={slices[6].data.url} alt={slices[6].data.alt} key=\\"img-\${i}\\" />"`;

exports[`@slicemachine/plugin-react #snippets snippet Link with useKey 1`] = `
"/* import { Link } from 'prismic-reactjs' */
<a key=\\"slices[7].data-\${i}\\" href={Link.url(slices[7].data)}>My Link</a>"
`;

exports[`@slicemachine/plugin-react #snippets snippet LinkToMedia with useKey 1`] = `
"/* import { Link } from 'prismic-reactjs' */
<a key=\\"slices[8].data-\${i}\\" href={Link.url(slices[8].data)}>My Link</a>"
`;

exports[`@slicemachine/plugin-react #snippets snippet Number with useKey 1`] = `"<span key=\\"slices[9].data-\${i}\\" >{ slices[9].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets snippet Select with useKey 1`] = `"<span key=\\"slices[10].data-\${i}\\" >{ slices[10].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets snippet StructuredText with useKey 1`] = `"<RichText render={slices[11].data} key=\\"rich-text-\${i}\\" />"`;

exports[`@slicemachine/plugin-react #snippets snippet Text with useKey 1`] = `"<span key=\\"slices[12].data-\${i}\\" >{ slices[12].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets snippet Timestamp with useKey 1`] = `"<span key=\\"slices[13].data-\${i}\\" >{ slices[13].data }>/span>"`;

exports[`@slicemachine/plugin-react #snippets snippet UID with useKey 1`] = `"<span>{ slices[14].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets supported type Boolean should be truthy 1`] = `"<span>{ slices[0].data ? 'true' : 'false' }</span>"`;

exports[`@slicemachine/plugin-react #snippets supported type Color should be truthy 1`] = `"<span style={{ color: slices[1].data }}>Some Text</span>"`;

exports[`@slicemachine/plugin-react #snippets supported type ContentRelationship should be truthy 1`] = `
"/* import { Link } from 'prismic-reactjs' */
<a href={Link.url(slices[2].data)}>My Link</a>"
`;

exports[`@slicemachine/plugin-react #snippets supported type Date should be truthy 1`] = `
"/* import { Date as ParseDate } from 'prismic-reactjs' */
<span>{ ParseDate(slices[3].data) }</span>"
`;

exports[`@slicemachine/plugin-react #snippets supported type Embed should be truthy 1`] = `
"// you might want to use a lib here (eg. react-oembed-container)
<div dangerouslySetInnerHTML={{ __html: slices[4].data }} />"
`;

exports[`@slicemachine/plugin-react #snippets supported type GeoPoint should be truthy 1`] = `"<span>{ slices[5].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets supported type Image should be truthy 1`] = `"<img src={slices[6].data.url} alt={slices[6].data.alt} />"`;

exports[`@slicemachine/plugin-react #snippets supported type Link should be truthy 1`] = `
"/* import { Link } from 'prismic-reactjs' */
<a href={Link.url(slices[7].data)}>My Link</a>"
`;

exports[`@slicemachine/plugin-react #snippets supported type LinkToMedia should be truthy 1`] = `
"/* import { Link } from 'prismic-reactjs' */
<a href={Link.url(slices[8].data)}>My Link</a>"
`;

exports[`@slicemachine/plugin-react #snippets supported type Number should be truthy 1`] = `"<span>{ slices[9].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets supported type Select should be truthy 1`] = `"<span>{ slices[10].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets supported type StructuredText should be truthy 1`] = `"<RichText render={slices[11].data}  />"`;

exports[`@slicemachine/plugin-react #snippets supported type Text should be truthy 1`] = `"<span>{ slices[12].data }</span>"`;

exports[`@slicemachine/plugin-react #snippets supported type Timestamp should be truthy 1`] = `"<span>{ slices[13].data }>/span>"`;

exports[`@slicemachine/plugin-react #snippets supported type UID should be truthy 1`] = `"<span>{ slices[14].data }</span>"`;

exports[`@slicemachine/plugin-react #story 1`] = `
Object {
  "data": "import MyComponent from '../slices/foo';
export default { title: 'FooBar' }

export const _DefaultSlice = () => <MyComponent slice={{\\"id\\":\\"default-slice\\",\\"name\\":\\"Default slice\\",\\"docURL\\":\\"...\\",\\"version\\":\\"sktwi1xtmkfgx8626\\",\\"description\\":\\"FooSlice\\",\\"primary\\":{\\"title\\":{\\"type\\":\\"StructuredText\\",\\"config\\":{\\"single\\":\\"heading1\\",\\"label\\":\\"Title\\",\\"placeholder\\":\\"This is where it all begins...\\"}},\\"description\\":{\\"type\\":\\"StructuredText\\",\\"config\\":{\\"single\\":\\"paragraph\\",\\"label\\":\\"Description\\",\\"placeholder\\":\\"A nice description of your feature\\"}},\\"\\":{\\"config\\":{\\"label\\":\\"\\",\\"constraint\\":{},\\"thumbnails\\":[]},\\"type\\":\\"Image\\"}},\\"items\\":{}}} />
_DefaultSlice.storyName = 'Default slice'

",
  "filename": "index.stories.js",
}
`;
